docker/build_artifacts/sagemaker/python_service.py

```diff
@@ -206,7 +208,11 @@ def _wait_for_model(self, model_name):
             while True:
                 time.sleep(0.5)
                 try:
-                    response = requests.get(url)
+                    session = requests.Session()
+                    retries = Retry(total=120,
+                                    backoff_factor=0.1)
```

Comment:
i don't think you want to retry 120 times with a `backoff_factor` of 0.1. From [here](https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html):

```
total – Total number of retries to allow. Takes precedence over other counts.
backoff_factor (float) – A backoff factor to apply between attempts after the second try...

    {backoff factor} * (2 ** ({number of total retries} - 1))
```

With a `backoff_factor` of 0.1, after the 10th retry, the amount of back-off/sleep on that iteration would be 102.4 seconds (subsequent retries, according to the docs, would actually exceed the `Retry.BACKOFF_MAX` of 120s and, therefore, would sleep for 120s for each of the remaining 110 retries). Example:

```
def test():
    session = requests.Session()
    retries = Retry(total=11, backoff_factor=0.1)
    session.mount("http://", requests.adapters.HTTPAdapter(max_retries=retries))
    start_time = datetime.now().timestamp()
    try:
        response = session.get("http://this-will-not-work")
    except:
        pass
    end_time = datetime.now().timestamp()
    print(f"slept a total of {end_time - start_time} seconds")

test()
slept a total of 206.1146810054779 seconds
```

Reply:
Yes, you are right.

To make the ``retries`` fit in the timeout limit (60s), the total should be 9: ``0.1 * (1 - 2 ^ n) / (1 - 2) <= 60``.

When total = 9, {end_time - start_time} is 51.1s, increasing total by 1 will exceed the timeout.